src/Core/Models/Data/EnvironmentUrlData.cs

```diff
@@ -20,5 +46,63 @@ public class EnvironmentUrlData
                 && string.IsNullOrEmpty(Notifications)
                 && string.IsNullOrEmpty(WebVault)
                 && string.IsNullOrEmpty(Events);
+
+        public Region Region
+        {
+            get
+            {
+                if (Base == Region.US.BaseUrl())
+                {
+                    return Region.US;
+                }
+                if (Base == Region.EU.BaseUrl())
+                {
+                    return Region.EU;
+                }
+                return Region.SelfHosted;
+            }
+        }
+
+        public EnvironmentUrlData FormatUrls()
+        {
+            return new EnvironmentUrlData
+            {
+                Base = FormatUrl(Base),
+                Api = FormatUrl(Api),
+                Identity = FormatUrl(Identity),
+                Icons = FormatUrl(Icons),
+                Notifications = FormatUrl(Notifications),
+                WebVault = FormatUrl(WebVault),
+                Events = FormatUrl(Events)
+            };
+        }
+
+        private string FormatUrl(string url)
+        {
+            if (string.IsNullOrWhiteSpace(url))
+            {
+                return null;
+            }
+            url = Regex.Replace(url, "\\/+$", string.Empty);
+            if (!url.StartsWith("http://") && !url.StartsWith("https://"))
+            {
+                url = string.Concat("https://", url);
+            }
+            return url.Trim();
+        }
+
+        public string GetDomainOrHost()
```

Comment:
⛏️ Just to be consistent with the helper naming I would change this to:
```suggestion
        public string GetDomainOrHostname()
```